ContactManagerApp/Pages/Index.cshtml.cs

```diff
@@ -19,73 +19,83 @@ public class IndexModel : PageModel
+    public List<Email> EmailAddresses { get; set; } = new List<Email>();
+
+    [BindProperty, Required, DisplayName()]
+    public List<Address> Addresses { get; set; } = new List<Address>();
+    */
     public IndexModel(ILogger<IndexModel> logger, IContactRepository db)
     {
         _logger = logger;
         _db = db;
     }
 
-    public async Task OnGetAsync()
+    public async Task OnGetAsync(CancellationToken cancellationToken)
     {
-        await LoadSampleData();
-        var contacts = _db.GetAllContacts();
+        try
+        {
+            await LoadSampleData(cancellationToken);
+            var contacts = _db.GetAllContacts();
+        }
+        catch (Exception e)
+        {
+            RedirectToPage("localhost:7296");
+        }
     }
     
-    private async Task LoadSampleData()
+    private async Task LoadSampleData(CancellationToken cancellationToken)
     {
-        if (_db.Count() == 0)
+        if (await _db.CountAsync(cancellationToken) == 0)
         {
             string file = System.IO.File.ReadAllText("sampledata.json");
             var contacts = JsonSerializer.Deserialize<List<Contact>>(file);
-            _db.AddRange(contacts);
-            _db.SaveChanges();
-        }
-        else
-        {
-            Contacts = await _db.ToListAsync();
+            await _db.AddRangeAsync(contacts, cancellationToken);
+            await _db.SaveChangesAsync(cancellationToken);
+            return;
         }
+            Contacts = await _db.ToListAsync(cancellationToken);
     }
 
-    [BindProperty, Required]
-    public string? FirstName { get; set; }
-
-    [BindProperty, Required]
-    public string? LastName { get; set; }
-
-    [BindProperty, Required]
-    public string? PhoneNumber { get; set; }
-
-    [BindProperty, Required]
-    public string? Birthday { get; set; }
-
-    /* [BindProperty, Required, DisplayName("Person's Email")]
-    public string? Email { get; set; }
-
-    [BindProperty, Required, DisplayName("Person's Address")]
-    public string? Address { get; set; } */
-
-    public async Task<IActionResult> OnPost()
+    public async Task<IActionResult> OnPost(CancellationToken cancellationToken)
     {
         if (ModelState.IsValid)
         {
-            Contact contact = new() { FirstName = FirstName!, LastName = LastName!,
-                PhoneNumber = PhoneNumber!, Birthday = Birthday! };
+            Contact contact = new() {
+                FirstName = FirstName!,
+                LastName = LastName!,
+                PhoneNumber = PhoneNumber!,
+                Birthday = Birthday!
+            };
+
             _db.Add(contact);
-            await _db.SaveChangesAsync();
+            await _db.SaveChangesAsync(cancellationToken);
             return RedirectToPage();
         }
         return Page();
     }
 
-    public IActionResult OnPostDelete(int id)
+    public IActionResult OnPostDelete(int id, CancellationToken cancellationToken)
```

Comment:
Http kennt verschiedeni methode. Haup methode isch GET und POST. es git aber au DELETE, PUT, PATCH und no paar anderi. 

In dem fall da würs warschinlich eher sin mache das als DELETE methode implementiere.

Da doku dezue: https://www.pragimtech.com/blog/blazor/delete-in-asp.net-core-rest-api/

Reply:
> Dennoch -- Delete ghört warschinlich eher in Update.cshtml als in index

wenn mer em CRUD prinzip folgt scho aber han da welle dass mer nöd muss uf eh wiiteri page gah zum en kontakt z'lösche. D'update page isch da demit ich au mal CRUD prinzip mittere wiiterfolgende page wie "/update" implementiert han wo erfolgriich öppis usfüehrt und wider uf die main index siite zrugg returned